File: src/Application Specific/Classes/Entity Collections/Order/ProductSupplierPriceRequest/ProductSupplierPriceRequestValidator.vb
Namespace Entity

Namespace ProductSupplierPriceRequests

Public Class ProductSupplierPriceRequestValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oProductSupplierPriceRequest As ProductSupplierPriceRequest)

        'make sure that contact object is valid
        If oProductSupplierPriceRequest.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oProductSupplierPriceRequest.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

                If oProductSupplierPriceRequest.OrderID = 0 Then
                    Me.AddCriticalMessage("Order missing")
                End If
                If oProductSupplierPriceRequest.SupplierID = 0 Then
                    Me.AddCriticalMessage("Supplier missing")
                End If
                If oProductSupplierPriceRequest.ProductID = 0 Then
                    Me.AddCriticalMessage("Product missing")
                End If
                If Not oProductSupplierPriceRequest.QuantityInPack > 0 Then
                    Me.AddCriticalMessage("Quantity must be greater than 0")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

    End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Order/ProductSupplierPriceRequest/ProductSupplierPriceRequestValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace ProductSupplierPriceRequests
    {
        // make sure that contact object is valid
        public class ProductSupplierPriceRequestValidator : BaseValidator
        {
            public void Validate(ProductSupplierPriceRequest oProductSupplierPriceRequest)
            {
                // make sure that contact object is valid
                if (oProductSupplierPriceRequest.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oProductSupplierPriceRequest.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oProductSupplierPriceRequest.OrderID == 0)
                {
                    AddCriticalMessage("Order missing");
                }

                if (oProductSupplierPriceRequest.SupplierID == 0)
                {
                    AddCriticalMessage("Supplier missing");
                }

                if (oProductSupplierPriceRequest.ProductID == 0)
                {
                    AddCriticalMessage("Product missing");
                }

                if (!(oProductSupplierPriceRequest.QuantityInPack > 0))
                {
                    AddCriticalMessage("Quantity must be greater than 0");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}